File: ecommerce-rest-api/src/main/java/com/example/ecommerce/service/PaymentService.java
package com.example.ecommerce.service;

import com.example.ecommerce.dto.payment.PaymentRequestDto;
import com.example.ecommerce.dto.payment.PaymentResponseDto;
import com.example.ecommerce.entity.enums.PaymentStatus;

import java.util.List;

/**
 * PaymentService
 *
 * Service layer for payment operations.
 */
public interface PaymentService {

    /**
     * Create payment record.
     */
    PaymentResponseDto create(
            PaymentRequestDto request
    );

    /**
     * Get payment by ID.
     */
    PaymentResponseDto getById(Long id);

    /**
     * Get all payments.
     */
    List<PaymentResponseDto> getAll();

    /**
     * Update payment.
     */
    PaymentResponseDto update(
            Long id,
            PaymentRequestDto request
    );

    /**
     * Delete payment.
     */
    void delete(Long id);

    /**
     * Process payment.
     */
    PaymentResponseDto processPayment(Long id);

    /**
     * Get payments by status.
     */
    List<PaymentResponseDto> getByStatus(
            PaymentStatus status
    );

    /**
     * Get payment by order ID.
     */
    PaymentResponseDto getByOrderId(Long orderId);
}

File: ecommerce-react-ui/src/services/paymentService.js
import api from "./api";

const paymentService = {
    getAllPayments: async () => {
        const response = await api.get("/payments");
        return response.data;
    },

    getPaymentById: async (id) => {
        const response = await api.get(`/payments/${id}`);
        return response.data;
    },

    getPaymentByOrderId: async (orderId) => {
        const response = await api.get(`/payments/order/${orderId}`);
        return response.data;
    },

    createPayment: async (payment) => {
        const response = await api.post("/payments", payment);
        return response.data;
    },

    updatePayment: async (id, payment) => {
        const response = await api.put(`/payments/${id}`, payment);
        return response.data;
    },

    deletePayment: async (id) => {
        const response = await api.delete(`/payments/${id}`);
        return response.data;
    },

    processPayment: async (id) => {
        const response = await api.post(`/payments/${id}/process`);
        return response.data;
    },
};

export default paymentService;